src/main/kotlin/org/rust/ide/refactoring/suggested/RsSuggestedRefactoringAvailability.kt

```diff
@@ -39,10 +41,19 @@ class RsSuggestedRefactoringAvailability(
     }
 
     private fun hasComplexChanges(
+        function: RsFunction,
         oldSignature: SuggestedRefactoringSupport.Signature,
         newSignature: SuggestedRefactoringSupport.Signature
-    ): Boolean = hasParameterAddedRemovedOrReordered(oldSignature, newSignature) ||
-        hasNameChanges(oldSignature, newSignature)
+    ): Boolean {
+        // Condition order is important here.
+        // hasTypeChanges cannot be called if parameters were removed or added
+        if (hasParameterAddedRemovedOrReordered(oldSignature, newSignature)) return true
+
+        // Type changes can only be observed by child method signatures
+        if (function.owner is RsAbstractableOwner.Trait && hasTypeChanges(oldSignature, newSignature)) return true
```

Comment:
`function.owner` uses resolve which may trigger `Slow operations are prohibited on EDT` exception. Let's use instead some light-weight approximate check like `function.parent?.parent !is RsImplItem`

Reply:
Damn, these blocking operations are real foot-guns. It almost asks for some `async/await` to be introduced here x)
Thanks, I changed it.